rust-runtime/aws-smithy-http-server-python/src/middleware/error.rs

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use pyo3::{exceptions::PyRuntimeError, PyErr};
+use thiserror::Error;
+
+/// Possible middleware errors that might arise.
+#[derive(Error, Debug)]
+pub enum PyMiddlewareError {
+    /// Returned when `next` is called multiple times.
+    #[error("next already called")]
+    NextAlreadyCalled,
+    /// Returned when request is accessed after `next` is called.
+    #[error("request is gone")]
+    RequestGone,
+    /// Returned when response is called after it is returned.
+    #[error("response is gone")]
+    ResponseGone,
```

Comment:
The comments are great - it feels like they should actually be used as error messages (by rephrasing them accordingly).